.github/ci_setup.sh

```diff
@@ -15,11 +15,14 @@ PLATFORM_NAME=$1
 
 # Set up the correct environment for the SYCL tests.
 if [ "${PLATFORM_NAME}" = "SYCL" ]; then
-   source /opt/intel/oneapi/setvars.sh --include-intel-llvm
+   if [ -f "/opt/intel/oneapi/setvars.sh" ]; then
+      source /opt/intel/oneapi/setvars.sh --include-intel-llvm
+   fi
    # Use clang/clang++ instead of icx/icpx, to avoid some aggressive math
    # optimizations that break some tests.
    export CC=`which clang`
    export CXX=`which clang++`
    export SYCLCXX="${CXX} -fsycl"
-   export SYCL_DEVICE_FILTER=host
+   export ONEAPI_DEVICE_SELECTOR=opencl:cpu
+   export SYCLFLAGS="-fsycl -fsycl-targets=spir64"
```

Comment:
The unfortunate thing is that this will now override the CUDA and ROCm SYCL targets in the new Docker images, so you'll need to find a different way to do this.

Reply:
Can I pass this as DSYCLFLAGS in the yaml?

Reply:
I think it's `-DCMAKE_SYCL_FLAGS`. But yes.

Reply:
That would technically work. But `ONEAPI_DEVICE_SELECTOR` is strictly an environment variable. We should not set a uniform value for all backends for that...

Reply:
But `ONEAPI_DEVICE_SELECTOR` is only for runtime, so that doesn't actually need to be changed as we are not actually running the CUDA + ROCm code (yet!).